lib/internal/test_runner/test.js

```diff
@@ -192,6 +199,17 @@ class Test extends AsyncResource {
       this.timeout = timeout;
     }
 
+    if (testNamePatterns !== null) {
+      const match = this instanceof TestHook || ArrayPrototypeSome(
+        testNamePatterns,
+        (re) => RegExpPrototypeExec(re, name) !== null
+      );
+
+      if (!match) {
+        skip = 'test name does not match pattern';
```

Comment:
how will this work with nested tests? I would expect a nested test to run even if its parent does not match the pattern

Reply:
I expect it to be used like `tap`'s `--grep` flag (although right now, passing regex flags is not handled):

```
Only run subtests tests matching the specified pattern.

Patterns are matched against top-level subtests in each
file. To filter tests at subsequent levels, specify
this option multiple times.

To specify regular expression flags, format pattern
like a JavaScript RegExp literal. For example: '/xyz/i'
for case-insensitive matching.

Can be set multiple times
```

Reply:
👍🏻 so this should probably be documented here as well.